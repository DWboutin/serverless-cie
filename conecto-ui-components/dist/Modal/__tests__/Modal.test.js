"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _react = _interopRequireDefault(require("react"));

var _enzyme = require("enzyme");

var _Modal = _interopRequireDefault(require("../../Modal"));

const handleClose = jest.fn();
global.console = {
  error: jest.fn()
};
describe('Modal', () => {
  it('should match snapshot when closed', () => {
    const wrapper = (0, _enzyme.shallow)(_react.default.createElement(_Modal.default, {
      isOpen: false,
      handleClose: handleClose
    }, "Test text here"));
    expect(wrapper).toMatchSnapshot();
  });
  it('should match snapshot when open', () => {
    const wrapper = (0, _enzyme.shallow)(_react.default.createElement(_Modal.default, {
      isOpen: true,
      handleClose: handleClose
    }, "Test text here"));
    expect(wrapper).toMatchSnapshot();
  });
  it('should have the prefix class', () => {
    const wrapper = (0, _enzyme.mount)(_react.default.createElement(_Modal.default, {
      isOpen: true,
      handleClose: handleClose
    }, "Test text here"));
    expect(wrapper.instance().classesPrefix).toEqual('to-modal');
    expect(wrapper.render().hasClass(wrapper.instance().classesPrefix)).toEqual(true);
    wrapper.unmount();
  });
  it('should call handleClose when click overlay', () => {
    const wrapper = (0, _enzyme.shallow)(_react.default.createElement(_Modal.default, {
      isOpen: true,
      handleClose: handleClose
    }, "Test text here"));
    wrapper.find('.to-modal-overlay').last().simulate('click');
    expect(handleClose).toBeCalled();
  });
});