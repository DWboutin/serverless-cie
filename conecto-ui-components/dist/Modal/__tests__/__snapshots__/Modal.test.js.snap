// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal should match snapshot when closed 1`] = `""`;

exports[`Modal should match snapshot when open 1`] = `
<Portal
  containerInfo={<body />}
>
  <div
    className="to-modal open"
  >
    <div
      className="to-modal-overlay"
      onClick={[MockFunction]}
    />
    <div
      className="to-modal-card"
    >
      <Button
        className="to-modal-card__close-btn"
        comp="button"
        handleClick={[MockFunction]}
        htmlType="button"
        type="menu"
      >
        <Icon
          className="active"
          type="menu"
        />
      </Button>
      <div
        className="to-modal-content"
      >
        Test text here
      </div>
    </div>
  </div>
</Portal>
`;
